pkgs/build-support/emacs/wrapper.nix

```diff
@@ -110,10 +101,10 @@ runCommand
         }
         mkdir -p $out/bin
         mkdir -p $out/share/emacs/site-lisp
-        ${optionalString nativeComp ''
+        ${lib.optionalString withNativeCompilation ''
```

Comment:
Could you use the `let inherit (lib) optionalString ...; in` pattern?

Reply:
One more contender for the Bikeshedding Battle!

https://github.com/NixOS/nixpkgs/pull/229030#discussion_r1187632385

Reply:
We're going to need an RFC on this, aren't we.

Reply:
As the `lib.` prefix doesn't really convey anything useful when calling `optionalString` I'd also prefer the `inherit` approach.
Either at the first scope where `optionalString` is used or at the top-level scope.

I don't have a super strong feeling on this and think that both prefixing and inherit is a massive improvement over using with.